### ISSUE TITLE
No error when setting invalid attribute pydantic v1

### ISSUE THREAD

**Issue by 0e4ef622** [microsoft/pyright#7183]
**Describe the bug**
There is no error when setting an unknown attribute on a Pydantic v1 model.

**Code or Screenshots**

```python
from pydantic import BaseModel

class Model(BaseModel):
    pass

v = Model()
v.another = "hi"  # no error
```

(pydantic version 1.10.14)

**VS Code extension or command-line**
CLI

```
$ pyright a.py
0 errors, 0 warnings, 0 informations
```
```
$ pyright --version
pyright 1.1.349
```
```
$ mypy a.py
a.py:7: error: "Model" has no attribute "another"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

For v2 there is also no error, but that seems to be a [pydantic issue](https://github.com/pydantic/pydantic/issues/8643).

**Comment by erictraut** [comment]
Pyright's behavior here is correct. The `BaseModel` class in pydantic defines a `__setattr__` method define as:

```python
class BaseModel:
     ...
    def __setattr__(self, name: str, value: Any) -> None: ...
```

That means all attribute assignments should be allowed by a type checker.

**Comment by erictraut** [comment]
I recently added support for `@no_type_check` in pyright. The decorator (according to PEP 484 and the typing spec) means that the function should be treated as though it has no type annotations. Since it already has no annotations, `@no_type_check` effective has no effect in this context.
